### PR TITLE
Support healthcare fhir new notifications config

### PR DESCRIPTION
**Pull request by joycezhou47**
Link: https://cloud.google.com/healthcare-api/docs/how-tos/pubsub#fhir_notifications_containing_fhir_resource_data
Fix: https://github.com/hashicorp/terraform-provider-google/issues/12061

If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [X] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-the-terraform-providers), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [X] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/main/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
healthcare: added `notification_configs ` field to `google_healthcare_fhir_store` resource
```

### CODE REVIEW

File: mmv1/products/healthcare/api.yaml [hunk 1]
```diff
@@ -379,6 +379,29 @@ objects:
                         resource is a recursive structure; when the depth is 2, the CodeSystem table will have a column called
                         concept.concept but not concept.concept.concept. If not specified or set to 0, the server will use the default
                         value 2. The maximum depth allowed is 5.
+      - !ruby/object:Api::Type::Array
+        name: notificationConfigs
```
Comment (c2thorn) [inline mmv1/products/healthcare/api.yaml:383]: As an array type, I believe this needs a description to build

**Review by c2thorn — CHANGES_REQUESTED** [review]

File: mmv1/products/healthcare/api.yaml (hunk 1 above)
Comment (joycezhou47) [reply mmv1/products/healthcare/api.yaml:383]: Hi Cameron,

Thanks for pointing this out. I am new here and I am still learning how to run tests, even how to use git properly. I'll come back to this PR once I figure out the proper workflow.

Comment (c2thorn) [reply mmv1/products/healthcare/api.yaml:383]: No worries, you've done well already! There's a lot of context to be learned. If you haven't seen it already, check out https://github.com/hashicorp/terraform-provider-google/blob/main/.github/CONTRIBUTING.md#tests for our testing guidelines

Comment (c2thorn) [reply mmv1/products/healthcare/api.yaml:383]: No, after this PR is merged you will not have to do any additional steps

**Review by c2thorn — COMMENTED** [review]
The test is failing on our end due to the following error:
```

    provider_test.go:315: Step 1/2 error: Error running apply: exit status 1
        
        Error: Error creating FhirStore: googleapi: Error 403: Permission pubsub.topics.publish denied on resource gs://tf-test-fhir-notifications<redactedsuffix> (or it may be malformed or not exist)
        
          with google_healthcare_fhir_store.default,
          on terraform_plugin_test.tf line 2, in resource "google_healthcare_fhir_store" "default":
           2: resource "google_healthcare_fhir_store" "default" {
```
Looks like some permission is needed, do you know if this is something we can add to the example?

**Comment by joycezhou47** [comment]
> The test is failing on our end due to the following error:
> 
> ```
> 
>     provider_test.go:315: Step 1/2 error: Error running apply: exit status 1
>         
>         Error: Error creating FhirStore: googleapi: Error 403: Permission pubsub.topics.publish denied on resource gs://tf-test-fhir-notifications<redactedsuffix> (or it may be malformed or not exist)
>         
>           with google_healthcare_fhir_store.default,
>           on terraform_plugin_test.tf line 2, in resource "google_healthcare_fhir_store" "default":
>            2: resource "google_healthcare_fhir_store" "default" {
> ```
> 
> Looks like some permission is needed, do you know if this is something we can add to the example?

The permission needed is the "pubsub.topics.publish" permission to the pubsub topic here. In order for the creation to actually succeed, that topic needs to exist and also the permission of that topic needs to be granted to the test project which is creating fhir store. Not sure if we can configure for this. 

(Also I noticed that I attached the "gs://" prefix which shouldn't be attached, now fixed.

**Comment by c2thorn** [comment]
Hi @joycezhou47, this is something we'll need to resolve for the test to pass. Can we grant permissions within the example using the [pubsub IAM controls](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/pubsub_topic_iam) ?

**Comment by joycezhou47** [comment]
pubsub permissions can be granted using IAM controls, but I need help to navigate the permission granting process for these tests since I don't really have an understanding of the test setup.

I see the "<redactedsuffix>" part on the failure message, I assumed that this means that each time the test runs, a pubsub topic is dynamically generated - this implies that we would also need to dynamically create this topic during the test.

However, if the topics are not dynamically generated, and if it should be much easier to configure - I'll need to find the project which this test is running against, and grant the pubsub.publish IAM permission to the service account of that project.

**Comment by c2thorn** [comment]
> pubsub permissions can be granted using IAM controls, but I need help to navigate the permission granting process for these tests since I don't really have an understanding of the test setup.
> 
> I see the "" part on the failure message, I assumed that this means that each time the test runs, a pubsub topic is dynamically generated - this implies that we would also need to dynamically create this topic during the test.

Yes these tests dynamically create and destroy resources each run, so the permissions will also need to be dynamic. I'm hoping something similar to https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/templates/terraform/examples/storage_pubsub_notifications.tf.erb is usable here

**Review by c2thorn — CHANGES_REQUESTED** [review]
Tests are good, thank you for the update. I think all this is now missing is some documentation about the datasource (html.markdown file in https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/website/docs/d) and a changelog note added about the datasource

**Comment by joycezhou47** [comment]
> 

Hi @c2thorn , I noticed that it's not the first time that cloud healthcare uses pubsub topic so I suspect that there's actually some setup that allows me to add the new field without doing the whole data source thing. 

Currently I can see one test failing, which is the VCR-test, is this test related the things I've added?

**Review by c2thorn — APPROVED** [review]
Test is unrelated, and this PR LGTM without the datasource. I'll approve and merge

**Pull request merged**